#VRML_SIM R2020a utf8
# license: Copyright Cyberbotics Ltd. Licensed for use only with Webots.
# license url: https://cyberbotics.com/webots_assets_license
# documentation url: https://www.cyberbotics.com/doc/guide/youbot
# The KUKA youBot is a powerful, educational robot that is especially designed for research and education in mobile manipulation, which counts as a key technology for professional service robotics. It consists of an omnidirectional platform, a five degree-of-freedom robot arm and a two-finger gripper.

PROTO Youbot [
  field SFVec3f    translation     0 0.12 0      # Is `Transform.translation`.
  field SFRotation rotation        1 0 0 -1.5708 # Is `Transform.rotation`.
  field SFString   name            "youBot"      # Is `Solid.name`.
  field SFString   controller      "youbot"      # Is `Robot.controller`.
  field SFString   controllerArgs  ""            # Is `Robot.controllerArgs`.
  field SFString   customData      ""            # Is `Robot.customData`.
  field SFBool     supervisor      FALSE         # Is `Robot.supervisor`.
  field SFBool     synchronization TRUE          # Is `Robot.synchronization`.
  field MFNode     bodySlot        []            # Extends the robot with new nodes in the body slot.
]
{
Robot {
  translation IS translation
  rotation IS rotation
  customData IS customData
  supervisor IS supervisor
  synchronization IS synchronization
  name IS name
  model "KUKA youBot"
  children [
    BodyMesh {
    }
    Group {
      children IS bodySlot
    }
    DEF WHEEL1 InteriorWheel {
      translation 0.228 -0.158 -0.055
      anchor 0.228 -0.158 -0.055
      name "wheel1"
      sensorName "wheel1sensor"
    }
    DEF WHEEL2 ExteriorWheel {
      translation 0.228 0.158 -0.055
      anchor 0.228 0.158 -0.055
      name "wheel2"
      sensorName "wheel2sensor"
    }
    DEF WHEEL3 ExteriorWheel {
      translation -0.228 -0.158 -0.055
      anchor -0.228 -0.158 -0.055
      name "wheel3"
      sensorName "wheel3sensor"
    }
    DEF WHEEL4 InteriorWheel {
      translation -0.228 0.158 -0.055
      anchor -0.228 0.158 -0.055
      name "wheel4"
      sensorName "wheel4sensor"
    }
    DEF PLATE Solid {
      translation -0.155 0 0
      children [
        PlateMesh {
        }
      ]
      name "plate"
      boundingObject Group {
        children [
          Transform {
            translation 0.008 0 0.045
            children [
              Box {
                size 0.25 0.17 0.007
              }
            ]
          }
          Transform {
            translation -0.015 0.112 0.045
            rotation 0 0 1 -0.13
            children [
              Box {
                size 0.17 0.08 0.007
              }
            ]
          }
          Transform {
            translation -0.015 -0.112 0.045
            rotation 0 0 1 0.13
            children [
              Box {
                size 0.17 0.08 0.007
              }
            ]
          }
          Transform {
            translation 0.076 0.084 0.045
            rotation 0 0 1 0.81
            children [
              Box {
                size 0.08 0.08 0.007
              }
            ]
          }
          Transform {
            translation 0.076 -0.084 0.045
            rotation 0 0 1 -0.81
            children [
              Box {
                size 0.08 0.08 0.007
              }
            ]
          }
        ]
      }
      physics Physics {
        density -1
        mass 0.5
      }
    }
  ]
  description "KUKA youBot"
  boundingObject Group {
    children [
      Transform {
        translation 0 0 -0.045
        children [
          Box {
            size 0.34 0.34 0.09
          }
        ]
      }
      Transform {
        translation 0 0 -0.045
        children [
          Box {
            size 0.56 0.23 0.09
          }
        ]
      }
    ]
  }
  physics Physics {
    density -1
    mass 22
    centerOfMass [
      0 0 -0.045
    ]
    inertiaMatrix [
      0.166204 0.418086 0.55459
      0 0 0
    ]
  }
  controller IS controller
  controllerArgs IS controllerArgs
}
}
